src/Raven.Server/Rachis/Follower.cs

```diff
@@ -60,11 +61,22 @@ private void FollowerSteadyState()
             
             while (true)
             {
-               
                 entries.Clear();
 
                 using (_engine.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
                 {
+                    using (context.OpenReadTransaction())
+                    {
+                        if (_engine.GetSnapshotRequest(context))
+                        {
+                            if (_engine.Log.IsInfoEnabled)
+                            {
+                                _engine.Log.Info($"{ToString()}: Snapshot was requested, so we close this follower.");
+                            }
+                            return;
```

Comment:
This happens on each and every read from the network, which seems excessive.
We need the state persistent, but we can do well by just checking a property on the engine, instead.

Reply:
done.